src/numeric_index.c

```diff
@@ -244,17 +191,17 @@ static void NumericRangeNode_Balance(NumericRangeNode **n) {
   (*n)->maxDepth = MAX((*n)->left->maxDepth, (*n)->right->maxDepth) + 1;
 }
 
-static NRN_AddRv NumericRangeNode_Add(NumericRangeNode **np, t_docId docId, double value) {
+static NRN_AddRv NumericRangeNode_Add(NumericRangeNode **np, t_docId docId, double value, size_t depth) {
   NumericRangeNode *n = *np;
   if (!NumericRangeNode_IsLeaf(n)) {
     // recursively add to its left or right child.
     NumericRangeNode **childP = value < n->value ? &n->left : &n->right;
-    NRN_AddRv rv = NumericRangeNode_Add(childP, docId, value);
+    NRN_AddRv rv = NumericRangeNode_Add(childP, docId, value, depth + 1);
 
     if (n->range) {
       // if this inner node retains a range, add the value to the range without
       // updating the cardinality
-      rv.sz += NumericRange_Add(n->range, docId, value, 0);
+      rv.sz += NumericRange_Add(n->range, docId, value);
       rv.numRecords++;
     }
 
```

Comment:
If we need to balance the sub-tree we are necessarily not currently looking at a node with a range, and the new root-to-be of the sub-tree also doesn't have a range.
I think this could work, but I worry about how clear and future-proof it will be.